### DISCUSSION TITLE
MQQT Delay?

### DISCUSSION THREAD

**Discussion by barkster** [aqualinkd/AqualinkD#158] [Q&A]
I was setting up my waterfall and I copied what was being used for my light and it has for state_topic: aqualinkd/Aux_4/delay but I wasn't really quite sure what the delay does.  I don't seem to see a delay when it turns on.  Can you tell me if I need this or not and what it actually does.  Pretty much everything has the delay except the heaters. Thank you.

- platform: mqtt
    unique_id: pool_waterfall
    name: "Pool Waterfall"
    state_topic: "aqualinkd/Aux_3"
    command_topic: "aqualinkd/Aux_3/set"
    json_attributes_topic: "aqualinkd/Aux_3/delay"
    json_attributes_template: "{{ {'active': value|int} | tojson }}"
    qos: 1
    payload_on: "1"
    payload_off: "0"
    retain: false
    icon: mdi:waterfall    

- platform: mqtt
  unique_id: pool_waterfall_delay
  state_topic: "aqualinkd/Aux_3/delay"
  name: "Waterfall Delay"
  qos: 0
  payload_on: "1"
  payload_off: "0"

**Comment by sfeakes** [comment]
You can’t set delay, only read that topic.  For a waterfall it’s meaningless.  It their for when you turn on something like pump and cleaner at the same time, but the cleaner will be on a delay for a few mins while the pump gets up to speed.  It’s set buy the control panel, so you can only read that topic.  BTW You also shouldn’t have delay topic on lights.

**Comment by barkster** [comment]
Thanks for reply, I just copied what was on this page https://github.com/sfeakes/AqualinkD/blob/master/extras/HASSIO.Implementation.txt and it has delay on light
